shared/fs/footer/container.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import * as Types from '../../constants/types/fs'
+import {compose, connect, setDisplayName, type Dispatch, type TypedState} from '../../util/container'
+import Footer, {type FooterProps} from './footer'
+import * as FsGen from '../../actions/fs-gen'
+
+const mapStateToProps = (state: TypedState) => ({
+  transfers: state.fs.transfers,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  opener: (p: Types.LocalPath) => dispatch(FsGen.createOpenLocalPathInFolder({localPath: p})),
+  dismisser: (key: string) => dispatch(FsGen.createDismissTransfer({key})),
+})
+
+const mergeProps = (stateProps, {opener, dismisser}, ownProps) =>
+  ({
+    downloads: Array.from(stateProps.transfers.filter(transferState => transferState.type === 'download'))
+      .sort(([_a, a], [_b, b]) => b.startedAt - a.startedAt) // newer first
+      .map(([key, transferState]) => ({
+        filename: Types.getLocalPathName(transferState.localPath),
+        completePortion: transferState.completePortion,
+        progressText: Math.round((1 - transferState.completePortion) * 4).toString() + ' s', // TODO: fix this when we have real estimate
+        isDone: transferState.isDone,
+        open: () => opener(transferState.localPath),
```

Comment:
you can have open just take this path so you don't need to make these closures for your rows

Reply:
You mean instead of `() => void`, `open` would be a type of `(path: Types.Path) => void`? I was passing that into `onClick` which expects a `() => void`, so that just means we need to make the same closure within `render()`. Is it the same or is one of them better?